New functionality for drop-down buttons on wxAuiToolBar items

carabsc opened the pull request:
This code allows for a new behavior when using drop-down buttons on toolbar items. Currently its assumed that when clicking on a item's drop-down button a menu will popup up to complete the operation, as implemented on the sample. These changes allow the use of generic transient popup windows that will popup when the drop-down button is clicked on and will close when the item's button is clicked on, without requiring a control on the transient window to complete the operation. The behavior is enabled with the style wxAUI_TB_ALLOW_CLICK_WITH_DROPDOWN and the click event handler will report whether the popup window was open or not when the item's button was pressed through the GetExtraLong() method.
The Aui sample was updated to showcase the new behavior.
NOTE: This code HAS NOT been tested on the MacOS port, but has been tested on the Windows and Linux ports.

carabsc committed:
Add a new behaviour to tools with dropdown buttons in wxAuiToolbar

Currently, when a tollbar is left-clicked inside a tool a dropdown event is fired
and if the event is skipped in an handler or not handled at all the code will fire
a menu event on left-up, signaling the click to the client code. If the event is
handled but not skipped then the click is discarded, assuming that whatever action
the click started will be finished on the popup.
This is suitable when a popup menu is used as dropdown, since a posterior click
on a menu item will complete the action. But when using a transient window with
some controls as a popup there needs to be an additional control on the popup
to complete the action. This change avoids that need.
The change implements a new style that allows the left-up code to respond as a
valid click even if the dropdown event was handled but not skipped, allowing
this sequence:

  1. The user clicks on the tool's dropdown; a popup shows up with controls
  2. The user modifies the state of the controls
  3. The user clicks on the tool again, closing the popup; the menu event
     handling code will be executed and the GetExtraLong method of the
     event wiil differentiate between a click on the button part of the
     tool and a click on the dropdown button

This allows the use of different code paths on the client code, depending
on whether the user clicked only on the button part of the tool or clicked
on the dropdown and then on the button part.

carabsc committed:
AUI demo modified to showcase commit 016df9690b63e85fa8fb6e95f419fa2bb0f3842d

carabsc committed:
Fixed double assignment bug on the AUI sample

carabsc committed:
Merge branch 'wxWidgets:master' into cara_changes

carabsc committed:
Added methods to get and set the disabled bitmap on wxAuiToolBar

carabsc committed:
Merge branch 'wxWidgets:master' into cara_changes

carabsc committed:
Updated the documentation of wxAuiToolBar to account for the style introduced on bc8c178e4f915e3b64e08d3c34559020bec16570

vadz commented:
Thanks for the PR, but I'm rather suspicious of all these `if ( HasCapture() )` checks. Could you please explain why were they added?

The new feature also needs to be documented and the build failures need to be fixed (AFAICS it's just a trivial warning about an unused parameter in the sample).

carabsc committed:
Added parameter to select the post-processing operation to apply to all bitmaps generated from a wxBitmapBundle containing SVG data.

When using bitmaps as icons in buttons or toolbars tipically an enabled and a disabled version of the same image are
required. The disabled version can be generated from the enabled one by using the ConvertToDisabled method, avoiding
the need to include two images per button in the executable.
When working with bitmap bundles this is more troublesome as the enabled bitmap needs to be created from bitmap data
first, the disabled bitmap has to be generated with the ConvertToDisabled method and then both bitmaps need to be
stored in separate bundles.
When working with SVGs even this is no longer possible as there's no method to modify the SVG data. And since the
buttons get the bitmap from the bundle for themselves, two versions of the same SVG need to be stored on the
executable. The new parameter allows the use of the factory methods like this:

	wxBitmapBundle active = wxBitmapBundle::FromSVG(data, wxSize(32, 32));
	All bitmaps generated from the SVG without post-processing

	wxBitmapBundle disabled = wxBitmapBundle::FromSVG(data, wxSize(32, 32), wxBitmapBundle::FromSVGDim);
	All bitmaps generated from this bundle will have their alpha channel at 33% of the original

	wxBitmapBundle active_dark = wxBitmapBundle::FromSVG(data, wxSize(32, 32), wxBitmapBundle::FromSVGInvertGreys);
	All bitmaps generated from this bundle will have their grey pixels inverted

	wxBitmapBundle disabled_dark = wxBitmapBundle::FromSVG(data, wxSize(32, 32), wxBitmapBundle::FromSVGDim |
                                                                                     wxBitmapBundle::FromSVGInvertGreys);
	All bitmaps generated from this bundle will have their alpha channel at 33% of the original
	and their grey pixels inverted

So now we can get four versions of the same SVG, saving up a lot of memory.

carabsc committed:
Merge branch 'cara_changes' of https://github.com/carabsc/wxWidgets into cara_changes

carabsc committed:
Merge branch 'wxWidgets:master' into cara_changes

carabsc commented:
Regarding the HasCapture calls they're used because the CaptureMouse call only happens if the wxEVT_AUITOOLBAR_TOOL_DROPDOWN event is not handled. Instead of having a boolean to keep track of that I've used the HasCapture to find out if that call ran, since each CaptureMouse call has to be paired with a single ReleaseMouse.
The CaptureMouse call comes from the original code and is used because the wxAuiToolBar has no actual buttons; its a single control that responds to mouse events and as such it may capture the mouse on a left mouse down event, releasing it on left mouse up.
Regarding the documentation, I can add some to the wxAuiToolBar class but will probably have to include some sample code to explain the functionality and that will take some time.
Regarding the build failures I'll have to look into that since I'm new to this Github stuff.

carabsc committed:
Added support for text colour in wxLogWindow messages.

Mappings between debug levels and colours can be manipulated with the methods

	AddLevelToColourMapping - Adds mapping between debug level and a colour
	RemoveLevelToColourMapping - Removes mapping
	ClearLevelToColourMapping - Clears all mappings

After setting up the appropriate mappings, specifying the debug level of the
message will also select the colour for its text shown in the wxLogWindow.

carabsc committed:
Split the wxAuiToolBarArt::UpdateColoursFromSystem method into two.

This facilitates the use of different bitmaps or bundles as drop down and overflow icons,
without needing to replicate code. To specify the bitmaps to use a sub-class of
wxAuiGenericToolBarArt should be created, with a constructor that sets the intended width
of the bitmaps by setting the m_overflowSize and m_dropdownSize variables and by overriding
the UpdateBitmapColoursFromSystem method and setting up the bitmaps into the
m_buttonDropDownBmp, m_disabledButtonDropDownBmp, m_overflowBmp and m_disabledOverflowBmp
variables. An instance of that class is then passed into the wxAuiToolBar as the art provider.

carabsc committed:
BUGFIX: Moved the UpdateColoursFromSystem call out of the wxAuiGenericToolBarArt constructor

The UpdateColoursFromSystem method is declared virtual but being called from the constructor
prevents virtual dispatching the call into a sub-class overriden method, forcing the user
code to call it if a sub-class is to be used as art provider. Now the call is done when
the sub-class is fed into the wxAuiToolBar as an art provider.

carabsc committed:
BUGFIX: Added actual support for hover bitmaps on wxAuiToolBar

Although it's possible to setup hover bitmaps for tool buttons they're never actually used.
These changes make it so a valid hover bitmap is used instead of the normal one if a button
is in an hover state, with the mouse over it.

carabsc committed:
Change on bc8c178e4f915e3b64e08d3c34559020bec16570; new event along with EVT_MENU on button click

The patch mentioned above required the use of the GetExtraLong() method of the wxCommandEvent
issued on a mouse click on the tool to find out if this click was done after using the dropdown
button to open an auxiliary window.
This change adds a new set of methods to the wxAuiToolBarEvent to report that information, named
IsButtonClickedAfterDropDown  and SetButtonClickedAfterDropDown, and issues an wxAuiToolBarEvent
right after the wxCommandEvent. Therefore now to detect button presses the used code can also
bind to wxAuiToolBarEvents with IDs of wxEVT_AUITOOLBAR_TOOL_BUTTON, if it wants to detect the
behaviour.

carabsc committed:
wxAuiToolBar: Button clicks are now masked by the DropDownVisibility flag

The mouse clicks on dropdown tools only generate events if the visibility
flag set by SetButtonDropDownVisibility allow them.

carabsc committed:
wxAuiToolBar: Extended the wxAuiToolBarDropDownDisplay enum to support disabling both buttons on a dropdown tool

Note that this is a separate flag, not related to the one set with EnableTool calls. Either of them can fully
disable a dropdown button but only the SetButtonDropDownVisibility call from the wxAuiToolBarItem class can
selectively disable parts of it.

carabsc committed:
BUGFIX: wxAuiGenericToolBarArt's ShowDropDown method takes into account the state of the tools in it

Once a tool is shown inside the overflow menu it's state is represented, in terms of being active.

carabsc committed:
BUGFIX on 65cef856c72d93d4732481bce2e53276852286e1: UpdateColoursFromSystem called from the wxAuiToolBar constructor

The wxAuiToolBar class constructor creates a default wxAuiGenericToolBarArt instance, to use as art provider. Since
the UpdateColoursFromSystem call was taken off the wxAuiGenericToolBarArt's constructor on the previous commit, it
must be added to wxAuiToolBar's constructor.

carabsc committed:
Changes on Aui sample to accomodate changes from fac992c99ae65f01d0ee341cfd9d36529ccc2153

carabsc committed:
wxAuiToolBar: Added emission of new event defined on fac992c99ae65f01d0ee341cfd9d36529ccc2153 for clicks on overflow menu items

carabsc committed:
Fix for build failure due to missing library in IPC sample

carabsc committed:
Added support for debug logging into a separate logging app, using IPC.

Using this facility allows the visualization of all debug messages
even if the debugged app becomes deadlocked.
Includes a sample server logging demo in the samples directory,
and client code for sending debug messages into it or a compatible
debug server.
Debug log entries can be nested by prefixing them with a two spaces
identation for each new sub-level. The entries can also be coloured.

carabsc committed:
Modified and added infrastructure files due to bakefile rerun

When adding the sample on 448c65f1d30a1e2510397ef85ce8a8c91003b769, as
per instructions, running the current 0.2.12 bakefile updated some and
created other files. That's the contents of this patch.

carabsc committed:
Compilation failed on MSW due to missing include

carabsc committed:
On the LogServer sample, allow copying messages to the clipboard

carabsc committed:
Fix on wxPopupWindow to allow full use of focusable controls on GTK+

Presently on GTK+ a POPUP_WINDOW is created when instantiating the
popup; but having focusable controls on it is not supported by GTK+,
as evidenced by the sample.
On the sample the text control on it can be modified by keyboard
input but will not show a caret while being modified, due to a GTK
limitation.
This fix instead creates a TOPLEVEL_WINDOW and hides it from the
pager and window bar, effectively allowing the focus to now be
represented with a caret on text controls.
Since the MSW port already had a style to support controls on the
popup this fix uses the same on GTK+, with the same meaning.
During testing it was shown that a TAB press would force the popup
to center on the screen so code was added to reposition the popup
when needed.

carabsc committed:
BUGFIX: Several bug fixes on the LogServer sample and the LogClient code

- Use a socket for communication instead of DDE, as DDE needs the message loop
  to be running on the client for processing messages, thereby negating the
  main usefulness of this code
- Fix the message column resizing on a LogServer window resizing, to keep the
  timestamp column at a fixed width and use the remainder for the message
- Add a critical section to the DoLogRecord method to prevent problems when
  posting messages from secondary threads.

carabsc committed:
BUGFIX on d66eec312a2ed97f307f695232be5afa734d75fe and original wxWidgets code

- Prevent a crash due to memory overwrite due to the incorrect looping through the bitmap data
- Account for the use of alpha premultiplication on all ports except wxGTK when post-processing
- When a dim bundle is requested the alpha values are now 50 % of the original, instead of 33.3 %
- The DoRasterize method was changed to avoid doing alpha premultiplication on the wxGTK port,
  as its not needed and if used will cause a dimming of any non-black pixels, as they both have
  an alpha value and scaled down colour components. This is actually a bug on the wxWidgets
  original code

carabsc committed:
BUGFIX: Change unconditional adding of path separator on wxGenericDirCtrl::FindChild method

If the path already has a trailing separator this bug causes a failure. The fix checks for
the trailing separator and only adds one if needed

carabsc committed:
CHANGE: Allow the use of the wxSplitterWindow class with non TLW

As mentioned on the code, I can't find a reason to restrict this class usage to TLWs

carabsc committed:
FEATURE: New virtual method added to wxComboCtrl to allow custom painting on the text entry
without a wxComboPopup control

The method below is called to render the text entry when a popup control is not used.
This change allows custom drawing the text entry on the wxOwnerDrawComboBox.

carabsc committed:
BUGFIX: Fix a compilation failure due to a missing forward declaration

carabsc committed:
BUGFIX: Missing changes on include file that should be on 54f24ecd96515ce0a0634aac793843fafe2f1719

carabsc committed:
FEATURE: Added a setter to the wxAcceleratorEntry to modify the command

carabsc committed:
FEATURE: Allow the use of custom colours to paint the selection on a wxListCtrl

Normally the system defined colours are used for painting the selected itens on a wxListCtrl.
This change allows using different foreground and background colours for it, conditionally.
The methods implemented for it are:

  void EnableAlternateSelectedColours(bool enable = true);

Choose which colours to use; system-wide (false) or custom (true)

  bool IsEnabledAlternateSelectedColours();

Check if custom colours are being used

  void SetAlternateSelectedTextColour(const wxColour& colour);
  void SetAlternateSelectedBackgroundColour(const wxColour& colour);

Set the foreground and background colours to use instead of the system defined ones

  wxColour GetAlternateSelectedTextColour();
  wxColour GetAlternateSelectedBackgroundColour();

Get the custom foreground and background colours used

carabsc committed:
FEATURE: Added a new style flag to the wxDataViewCtrl class to prevent interactive searching on wxGTK

As implemented, the wxDataViewCtrl enters in an interactive search mode whenever a keypress is not
handled by the main control. That causes a popup text entry control to be shown below the main control,
to no purpose as the main control does nothing with it.
Adding the new wxDV_PREVENT_SEARCH style flag will prevent that behaviour on wxGTK and will do nothing
on all other ports.

carabsc committed:
CHANGE: Changes to the wxAuiToolbar and associated headers

- The wxAuiPaneButtonState enumeration is moved from the framemanager.h include file into the
  auibar.h include file, since that's where the values are referenced
- The wxAuiPaneButtonState enumeration is expanded to include some new private states
- The API for managing drop down buttons was changed to include these methods:

  void SetToolSplitState(int toolId, bool b);

Splits the state of the main button and the drop down button on the tool identified by toolId,
depending on b

  void SetToolButtonEnabled(int toolId, bool b);
  void SetDropDownEnabled(int toolId, bool b);

Set the state of the individual buttons on the tool identified by toolId, if split previously

  bool GetToolButtonEnabled(int toolId) const;
  bool GetToolDropDownEnabled(int toolId) const;

Gets the state of the individual buttons on the tool identified by toolId

  void CacheToolState();
  void ClearToolStateCache();

Helper methods for storing and clearing the cached state of a tool

  void RefreshHover();

Helper method to repaint the tool on which the mouse is hovering

carabsc committed:
Additional to 57a7b220d46ad3096a3a13c3b6e35dd51580c3c0

carabsc committed:
wxGenericListCtrl: Allow the use of wxLC_VIRTUAL with all other styles; paint optimization

-Extends the virtual mode to all styles of lists, as supported by wxListCtrl on MSW, allowing the
 access to the list data to be done through the existing API implemented for the wxLC_REPORT style.
 This allows much faster interactive operation when working with thousands of items.
-Preserves the wxLC_VIRTUAL style when changing between styles, using the SetSingleStyle call,
 which previously was only done on MSW.
-Decoupled the number of defined columns from the style selected. This allows defining columns when
 instatiating the control and then using the SetSingleStyle call to change between styles on the fly.
 Previously it was assumed that columns were only defined when using the wxLC_REPORT style.
-Optimization on the paint code, avoiding rendering all items when not in wxLC_REPORT style. The
 code now uses the calculated geometry to filter out the items that must be rendered, which can amount
 to a few dozens in a set of thousands of items. This speeds up the paint code in styles other than
 wxLC_REPORT, regardless of the use of wxLC_VIRTUAL, significantly speeding up scrolling.

carabsc committed:
BUGFIX: Added missing mapping from WXK_NUMPAD_ENTER to VK_RETURN on MSW

carabsc committed:
BUGFIX: Mouse capture not released on wxAuiManager

Should a pane resize fail on DoEndResizeAction the mouse was recaptured while discarding the operation to perform.
This caused a host of UI problems as the mouse was not released even after LeftUp and eventually an app crash as
a posterior LeftDown over a sizer would try to recapture the mouse, failing on an assert.
This bug could be triggered by trying to resize a pane with no non-fixed one to the right on the dock,
i.e. a left dock with two resizable panes and a fixed one, from top to bottom, trying to move the last sizer.

carabsc committed:
BUGFIX: wxAuiManager - fix the positioning/sizing of fixed panes

A fixed panes must have a zero proportion, so as to not participate in the calculations,
regardless of having a defined minimum size or not.

carabsc committed:
FEATURE: Add info field to wxAuiNotebook pages, with accompaning methods

carabsc committed:
Update wxAuiToolbar code to use native rendering

carabsc committed:
Add optional support for hover and narrower expander button cells in wxPropertyGrid

carabsc committed:
Add methods to draw tree item backgrounds, toolbar buttons and tabs in wxNativeRenderer

carabsc committed:
Add menu entries to Delete and Select All in the wxTextCtrl's popup menu

carabsc committed:
Add optional hover support to the generic implementation of wxDataViewCtrl

carabsc committed:
Refactoring of wxAuiNotebook to add support for left and right alignment and native rendering

- Support for left and right aligned tabs
- Sample modified accordingly
- Support for native rendered tabs; for this purpose the API between wxAuiTabContainer and
  wxAuiTabArt has been modified by changing the DrawTab signature and adding a DrawTabContent
  method, which should contain the code to draw the inside of each tab while the code to draw
  outline of the tab should be kept on DrawTab. This is needed because the native rendering
  methods only support rendering the outline on the top orientation; so graphic operations
  need to be performed to draw those into the tab control in the correct orientation, which
  can be different from the orientation of the tab contents.

carabsc committed:
Update wxAuiMSWTabArt to match the API changes on 2efd6061fd6c17068110a8c82181faf9d83d95e1

carabsc committed:
More changes related to 2efd6061fd6c17068110a8c82181faf9d83d95e1

carabsc committed:
Optimized painting on hover for the generic implementation of wxDataViewCtrl

carabsc committed:
Added hovering support on mouse move on wxPropertyGrid. Also modified the look to approximate the tree controls,
by reducing the spacings on the label column.

carabsc committed:
Added hovering support on mouse move for wxListCtrl, enabled with the style wxLC_HOVER_HIGHLIGHT.

carabsc committed:
Added support for another background style on wxAuiwxAuiGenericToolBarArtToolbar, enabled with the
wxAUI_TB_PARENT_BACKGROUND style. Using this style the toolbar's background will use the colour of
the containing window's background as a flat colour.
Also prevents the aquisition of focus by the toolbar, since no keyboard navigation is supported.

carabsc committed:
Fixes on wxGenericComboCtrl to make it have a more native look-and-feel.

carabsc committed:
Changes to wxRenderedNative class

- Added method GetButtonPaddingSize to return the padding around the bitmap inside a button.
  This method can be used to calculate the bitmap size that fits without scaling inside a
  button of a defined size.
- Changes to wxRendererXP::DrawTreeItemButton to better support hovering.
- Changes to wxRendererXP::DrawTabControlTab to use device independant margins.

carabsc committed:
Changes to the wxTreeCtrl:

- Method ScrollTo is modified to accept an optional 'to_top' parameter. Currently the method  will make sure
  the item selected becomes visible but gives no assurances on its position on the visible structure; by
  passing 'true' on this parameter the selected item will be shown on the top of the visible structure, if at
  all possible.
- Added support for hovering on mouse moves, enabled with the style wxTR_HOVER_HIGHLIGHT.
- Added support for making the in-place editor as wide as the tree control, enabled with the style
  wxTR_FULL_WIDTH_EDITOR.
- When the label text is being changed with the in-place editor the new wxEVT_TREE_DOING_LABEL_EDIT event is
  issued.
- Fix a bug that prevents changes to the foreground and background colours of the in-place editor on MSW.

carabsc committed:
Added two new icons to wxArtProvider to represent favorites.

carabsc committed:
Changes to wxGenericDirCtrl:

- Added support for highlighting the entire selected row, instead of just the text, that can be enabled
  with the wxDIRCTRL_ROW_HIGHLIGHT style flag.
- Added support for themed representation, that can be enabled with the wxDIRCTRL_THEMED style flag.
- Added support for hovering on mouse moves, enabled with the wxDIRCTRL_HOVER_HIGHLIGHT style flag.
- Added support for showing and manipulating bookmarks / favorites, to be displayed in a separate section
  of the tree control at the top. This is enabled with the wxDIRCTRL_BOOKMARKS style flag.
- Added support for having label in-place editors with a width as wide as possible. This support can be
  enabled with the wxDIRCTRL_FULL_WIDTH_EDITOR style flag.
- Added four new events:

	wxEVT_DIRCTRL_DIRACTIVATED - A directory was activated (typically double-clicked on)
	wxEVT_DIRCTRL_FAVACTIVATED - A bookmark / favorite was activated (see above)
	wxEVT_DIRCTRL_FAVEDITED - The name of a bookmark / favorite was edited.
	wxEVT_DIRCTRL_EDITING - The name of a bookmark / favorite is being edited.

carabsc committed:
Added support for manually resizing a popup window, using the wxPU_ALLOW_RESIZING style flag

carabsc committed:
BUGFIX: On wxSplitterWindow allow the use a secondary window without a primary one

carabsc committed:
Fix the wxStyledTextCtrl autocompletion popup

The changes implemented on the wxPopupWindow have broken the autocompletion popup of this control.
The fix simply changes the mode of operation of the popup window, avoiding the breakage. I also
don't see why the popup window was created with a wxPU_CONTAINS_CONTROLS style flag since the
list control does not accept focus.

carabsc committed:
BUGFIX: Prevent popup invocation on Space keypress on wxGenericComboCtrl

Previously I added this test but forgot that the user may want to enter a space into the control,
which would cause the invocation instead of simply adding the character.

carabsc committed:
BUGFIX: Prevent crash on wxListCtrl when hovering

carabsc committed:
REVERT: Revert changes on fac992c99ae65f01d0ee341cfd9d36529ccc2153 and bc8c178e4f915e3b64e08d3c34559020bec16570

Those changes implemented a detection for the clicking procedure on dropdown tools in wxAUIToolbar, that is best
performed using facilities now available on the wxPopupWindow class, namely event information stored on a left
mouse click. The AUI sample should be fixed to work that way too.

carabsc committed:
BUGFIX: On a wxAUIToolbar with a minimal size the last tool is not rendered nor accessible

When painting and when fit testing the code mishandles the border case of the last tool on the toolbar
aligning visually its rightmost (horizontal toolbar) or bottommost (vertical toolbar) edge with the edge
of the toolbar, causing that last tool to not be shown nor clickable.

carabsc committed:
wxHTMLWindow: Promote the ScrollToAnchor method to public

Promoting this method allows scrolling into view of anchors to be controlled programatically in user code

carabsc committed:
Added a new flag to decouple the rendering of the label from the content of a property on wxPropertyGrid

Previously if a property was disabled its label was shown visually as disabled. A new constructor flag
called wxPG_LABELS_ALWAYS_ENABLED is introduced to avoid that behaviour.

carabsc committed:
Change to the wxRendererGeneric::DrawHeaderButtonContents method to allow specifying the bitmap position inside the button

carabsc committed:
Allow setting the bitmap relative position inside a wxHeaderColumn

carabsc committed:
Allow bitmaps inside the wxGrid column headers with a defined alignment

New wxGridTableBase methods:

GetColLabelBitmap
GetColLabelBitmapBundle
  Define the bitmap / bitmap bundle to use on the column header

GetColLabelAlignment
  Define the alignment of the bitmap within the column header

carabsc committed:
BUGFIX: On wxAuiToolbar tools that do not fit are shown bunched up together at the end

The 3bb4e68f02ca74d530a8c4a875119cd16a87e949 commit solved one problem introducing another;
if a tool did not fit in the toolbar it was rendered at the end of the toolbar nevertheless,
because it used the size computed by the sizer to check for fit but that size could become
zero once the tools exausted all the available space in the toolbar.
This fix used the dimensions provided by the art provider for the fit check.

vadz commented:
I'm sorry, I'm not really sure what's going on with this PR but I think the only good way forward is to split it in separate parts and resubmit because it's just impossible to understand anything here any more.

carabsc commented:
That PR was superseded by several other commits. Right now I have around 70
commits in my fork that fix bugs, implement missing or new functionalities
and improve several others. Since I haven't merged with upstream since the
fork some commits may need rework to merge properly.
For now unless someone wants to cherrypick commits and apply them the best
course will be to ignore the PR, as right now I don't have time to merge
the upstream with my fork.
Em 29/02/2024 02:35, "VZ" ***@***.***> escreveu:

> I'm sorry, I'm not really sure what's going on with this PR but I think
> the only good way forward is to split it in separate parts and resubmit
> because it's just impossible to understand anything here any more.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/wxWidgets/wxWidgets/pull/22579#issuecomment-1970291710>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AWDADQGDHIDE7TGMCH6BO33YV2JWZAVCNFSM52B4XEE2U5DIOJSWCZC7NNSXTN2JONZXKZKDN5WW2ZLOOQ5TCOJXGAZDSMJXGEYA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

vadz commented:
It looks like there could be useful things here, but it's impossible to apply it as is, so I have no choice but to close it.
